src/Eval/External.ml

```diff
@@ -55,8 +55,12 @@ let int64_cmpop op = int64_fun2 (fun x y -> of_bool (op x y))
 
 let str_cmpop op = str_fun (fun s1 -> str_fun (fun s2 -> of_bool (op s1 s2)))
 
+let runtime_error msg =
+  failwith ("Runtime error: " ^ msg)
```

Comment:
I don't think this should be implemented using `failwith`. Triggering this function will result in output:
```
Fatal error: exception Failure("Runtime error: Assertion failed")
```
Personally i think it should be Just `Runtime error: Assertion failed`, so this code should be changed to eprintf + exit 1

Reply:
I would add possibility to add custom error code/number on exit.

Reply:
I agree that this error should not be implemented with `failwith`. Maybe a custom exception that is handled in the main file would be better? The REPL implementation could catch this exception and recover with a prompt for an another command, instead of aborting the interpreter.

Reply:
Actually that would probably be best

Reply:
I don't know and possible I am wrong but this sounds like an error implementation which we would make in standard library.

Reply:
Do you mean the issue #130? I think these two are two different things. In this PR we talk about errors that should never occur in correct programs (but unfortunately, most of programs are incorrect), so the effect of such error is pure (`[]`). In #130 the error has some effect visible in types. However, after merging this PR, we may rethink if we still want to implement #130.

Reply:
However, I realized that catching runtime error in REPL is not as trivial as I thought. It is not just adding extra match clasue in [Eval.ml](https://github.com/fram-lang/dbl/blob/f54c429d0f20614d0d91a616908f7cbd329ce1e6/src/Eval/Eval.ml#L118), because the exception are catch before the evaluation. I think it is still possible to handle it correctly, because the evaluator is in CPS, but it would be a more complex change, and should be done in a separate issue.